lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -457,6 +472,38 @@ withSizeHeader16be x = do
     putWord16be (fromIntegral $ BS.length bs)
     putByteString bs
 
+-- | Convert an 'Integral i' to 'Int'. Doesn't /expect/ overflow, but uses
+-- 'overflowSafeCoerce' internally.
+toInt :: (Show i, Typeable i, Integral i) => i -> Int
+toInt = overflowSafeCoerce
+
+-- | Like 'fromIntegral' but throws an exception if the conversion would have
+-- caused overflow.
+--
+-- > f = overflowSafeCoerce @Int @Word8
+-- > f 0
+-- 0
+-- > f 255
+-- 255
+-- > f 256
+-- *** Exception: overflowSafeCoerce: cannot convert (256 :: Int) to Word8
+overflowSafeCoerce
```

Comment:
We could move this to like `Cardano.Wallet.Unsafe` and test it?

Also is the name and/or the function in general sound?

Reply:
> Also is the name and/or the function in general sound?

No :sweat_smile: ... I think there's a misunderstanding about what `safe` and `unsafe` means in Haskell. This is an example of a very unsafe function. A safe function doesn't throw at runtime, but instead, captures the failure in its type. For example:

```hs
safeHead :: [a] -> Maybe a
```

in contrast with `head`, which should really be called `unsafeHead :: [a] -> a`.

Reply:
Also, since we aren't really `coercing` here but using `fromIntegral`, I'd suggest calling it `unsafeFromIntegral`.
Beside, I am not sure that this function buys us _much_ compare to using `toEnum :: Int -> x`

```
λ> toEnum @Word8 444
*** Exception: Enum.toEnum{Word8}: tag (444) is outside of bounds (0,255)
```

The only gain is maybe in the name (if it were called `unsafeXXX`) to remind us that some unsafe coercions are going on. Not sure that is very useful; We should definitely switch `fromIntegral` usafe to `toEnum` (because `fromIntegral` actually overflows and "loops" back to `minBound`). But we don't have to define this function since it already exists!